### ISSUE TITLE
Build failure: cargo-outdated 0.14.0 on aarch64-darwin

### ISSUE THREAD

**Issue by badboy** [NixOS/nixpkgs#265642]
### Steps To Reproduce

1. build `cargo-outdated`

### Build log

Important bits:

```
note: ld: framework not found CoreServices
```

Longer log:

```
error: builder for '/nix/store/5g97wlsx47axkw004nrlcr46b4yi4iar-cargo-outdated-0.14.0.drv' failed with exit code 101;
       last 10 log lines:
       >    Compiling git2-curl v0.18.0
       >    Compiling cargo-outdated v0.14.0 (/private/tmp/nix-build-cargo-outdated-0.14.0.drv-0/cargo-outdated-0.14.0.tar.gz)
       > error: linking with `/nix/store/sa6hywsm1mqfyd1xakyzv4ljjsb3hawh-clang-wrapper-11.1.0/bin/cc` failed: exit status: 1
       >   |
       >   = note: LC_ALL="C" PATH="/nix/store/8zq6sng72fx8hpnmcxwfgvr9vx4r1j5f-rustc-1.72.1/lib/rustlib/aarch64-apple-darwin/bin:/nix/store/4qdazhg6d6ck4ry9wh099f25hfgkpg5k-cargo-1.72.1/bin:/nix/store/yvljjfcbmk1grdjnplyyjymz642khbsm-cargo-auditable-0.6.1/bin:/nix/store/qn4rg6zdcfc6jb8d0yiwa30w7k2dscvv-pkg-config-wrapper-0.29.2/bin:/nix/store/62p2hzr9s4hpimwfpzd9m6ca3zlyvhm0-auditable-cargo-1.72.1/bin:/nix/store/4qdazhg6d6ck4ry9wh099f25hfgkpg5k-cargo-1.72.1/bin:/nix/store/8zq6sng72fx8hpnmcxwfgvr9vx4r1j5f-rustc-1.72.1/bin:/nix/store/sa6hywsm1mqfyd1xakyzv4ljjsb3hawh-clang-wrapper-11.1.0/bin:/nix/store/ljz45vk739778lw2xaw2jvy7ihip2q07-clang-11.1.0/bin:/nix/store/943sxl4vcfpfg6xaagxvgwbgz9scl7lc-coreutils-9.3/bin:/nix/store/m488d5iwzn93bdk1j5gxl77k3zb8y285-cctools-binutils-darwin-wrapper-11.1.0-973.0.1/bin:/nix/store/x5bm27csww06snmzrsqhxfzl0i8mad5c-cctools-binutils-darwin-11.1.0-973.0.1/bin:/nix/store/943sxl4vcfpfg6xaagxvgwbgz9scl7lc-coreutils-9.3/bin:/nix/store/ngyiha629vs37hb2pwmk489ndf7nmh20-findutils-4.9.0/bin:/nix/store/d1gsih58dvmj2da2im0c5k3rz2x6zhff-diffutils-3.10/bin:/nix/store/fh1327f06qipzcnv2di22vssbbzk1hrz-gnused-4.9/bin:/nix/store/kxbn7dv0i7p0ylsbr53j7z0gf45kim78-gnugrep-3.11/bin:/nix/store/qllkm9swhw8fgs46v2mk64kl88cm7jh8-gawk-5.2.2/bin:/nix/store/fpmgc7y2dihni7kpwvgaq46fjcds6my1-gnutar-1.35/bin:/nix/store/5dyk8m2il19cm2q7bvrlvnd6957jpp6s-gzip-1.13/bin:/nix/store/722g4vgl3g935h805y7phifmdvq2xl8p-bzip2-1.0.8-bin/bin:/nix/store/5sbqambqp3w2sc5mm3b7ddgla5jp7v9a-gnumake-4.4.1/bin:/nix/store/zzpm4317hn2y29rm46krsasaww9wxb1k-bash-5.2-p15/bin:/nix/store/v2s7n510v8zf67ngj4962zjia6h25r88-patch-2.7.6/bin:/nix/store/w2hc8y9m36d2nwrvrf27yrvzpwc8pbcm-xz-5.4.4-bin/bin:/nix/store/ps2jbdzj9kxch1b77apis0i06chk91qw-file-5.45/bin" VSLANG="1033" ZERO_AR_DATE="1" "/nix/store/sa6hywsm1mqfyd1xakyzv4ljjsb3hawh-clang-wrapper-11.1.0/bin/cc" "-arch" "arm64" "/private/tmp/nix-build-cargo-outdated-0.14.0.drv-0/rustcUk0U34/symbols.o" "/private/tmp/nix-build-cargo-outdated-0.14.0.drv-0/cargo-outdated-0.14.0.tar.gz/target/aarch64-apple-darwin/release/deps/cargo_outdated-20d94ae7db0e29cb.cargo_outdated.1e33c52461b11aea-cgu.03.rcgu.o" "-L" "/private/tmp/nix-build-cargo-outdated-0.14.0.drv-0/cargo-outdated-0.14.0.tar.gz/target/aarch64-apple-darwin/release/deps" "-L" "/private/tmp/nix-build-cargo-outdated-0.14.0.drv-0/cargo-outdated-0.14.0.tar.gz/target/release/deps" "-L" "/nix/store/sa6hywsm1mqfyd1xakyzv4ljjsb3hawh-clang-wrapper-11.1.0/resource-root/lib/darwin" "-L" "/private/tmp/nix-build-cargo-outdated-0.14.0.drv-0/cargo-outdated-0.14.0.tar.gz/target/aarch64-apple-darwin/release/build/curl-sys-1579647339e9def0/out/build" "-L" "/private/tmp/nix-build-cargo-outdated-0.14.0.drv-0/cargo-outdated-0.14.0.tar.gz/target/aarch64-apple-darwin/release/build/libnghttp2-sys-5cdfe1edc47dca7e/out/i/lib" "-L" "/nix/store/ycsnhmhjj1xwgx2ljqxnifgc8id3y4yy-zlib-1.3/lib" "-L" "/private/tmp/nix-build-cargo-outdated-0.14.0.drv-0/cargo-outdated-0.14.0.tar.gz/target/aarch64-apple-darwin/release/build/libgit2-sys-7c07d45482e9baf4/out/build" "-L" "/private/tmp/nix-build-cargo-outdated-0.14.0.drv-0/cargo-outdated-0.14.0.tar.gz/target/aarch64-apple-darwin/release/build/libssh2-sys-3fe97e2c875682fb/out/build" "-L" "/opt/homebrew/opt/openssl@3/lib" "-L" "/nix/store/8zq6sng72fx8hpnmcxwfgvr9vx4r1j5f-rustc-1.72.1/lib/rustlib/aarch64-apple-darwin/lib" "/private/tmp/nix-build-cargo-outdated-0.14.0.drv-0/rustcUk0U34/liblibgit2_sys-8948c6c9994bc90d.rlib" "/private/tmp/nix-build-cargo-outdated-0.14.0.drv-0/rustcUk0U34/liblibssh2_sys-97564b0d060ead8b.rlib" "/private/tmp/nix-build-cargo-outdated-0.14.0.drv-0/rustcUk0U34/libcurl_sys-fae43e42c2a3a6c2.rlib" "/private/tmp/nix-build-cargo-outdated-0.14.0.drv-0/rustcUk0U34/liblibnghttp2_sys-05f4832c7384b27c.rlib" "/nix/store/8zq6sng72fx8hpnmcxwfgvr9vx4r1j5f-rustc-1.72.1/lib/rustlib/aarch64-apple-darwin/lib/libcompiler_builtins-1f437443701d86bd.rlib" "-liconv" "-framework" "Security" "-framework" "CoreFoundation" "-lssl" "-lcrypto" "-lclang_rt.osx" "-framework" "Security" "-framework" "CoreFoundation" "-framework" "CoreServices" "-framework" "SystemConfiguration" "-lz" "-framework" "CoreFoundation" "-liconv" "-lSystem" "-lc" "-lm" "-L" "/nix/store/8zq6sng72fx8hpnmcxwfgvr9vx4r1j5f-rustc-1.72.1/lib/rustlib/aarch64-apple-darwin/lib" "-o" "/private/tmp/nix-build-cargo-outdated-0.14.0.drv-0/cargo-outdated-0.14.0.tar.gz/target/aarch64-apple-darwin/release/deps/cargo_outdated-20d94ae7db0e29cb" "-Wl,-dead_strip" "-nodefaultlibs" "/private/tmp/nix-build-cargo-outdated-0.14.0.drv-0/cargo-outdated-0.14.0.tar.gz/target/aarch64-apple-darwin/release/deps/cargo_outdated_audit_data.o" "-Wl,-u,_AUDITABLE_VERSION_INFO"
       >   = note: ld: framework not found CoreServices
       >           clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
```

### Additional context

Updated here https://github.com/NixOS/nixpkgs/pull/264981
Commit: https://github.com/NixOS/nixpkgs/commit/0543391e4eb9c3a30405285902bd2015fc64ac79
Failure listed in the run too: https://github.com/NixOS/nixpkgs/runs/18300185436

### Notify maintainers

cc @ivan @matthiasbeyer

### Metadata

```console
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.6.0, macOS 13.5.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.17.0`
 - nixpkgs: `/nix/store/8kfifw0fxdhgsas8ywy9jikzgz2h2aa4-source`
```

**Comment by FliegendeWurst** [comment]
PR with fix: #265643

Created 1 minute after your report :rocket:

**Comment by badboy** [comment]
Fixed by #265643
